pkg/sharings/recipients.go

```diff
@@ -209,6 +206,20 @@ func (rs *RecipientStatus) Register(instance *instance.Instance) error {
 	return nil
 }
 
+func getPublicName(instance *instance.Instance) (string, error) {
```

Comment:
If it's ok to return a default public name in the general case, no problem to make it part of `Instance`